documentation/src/docs/beginner_dev_guide.md

```diff
@@ -0,0 +1,200 @@
+# 4.1 Create a Pravega Stream
+
+Let’s first get to know Pravega’s client APIs by creating a stream with a fixed scaling policy of 1 segment. We’ll need a StreamConfiguration to define this:
+```java
+StreamConfiguration streamConfig = StreamConfiguration.builder()
+        .scalingPolicy(ScalingPolicy.fixed(1))
+        .build();
+```
+With this streamConfig, we can create streams that feel a bit more like traditional append-only files. Streams exist within scopes, which provide a namespace for related streams. We use a [StreamManager](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/admin/StreamManager.html) to tell a Pravega cluster controller to create our scope and our streams. Since we are using a standalone Pravega let's use the below controller address.
+```java
+URI controllerURI = URI.create("tcp://localhost:9090");
+```
+The code to create a Pravega stream is as follows
+```java
+try (StreamManager streamManager = StreamManager.create(controllerURI)) {
+        streamManager.createScope("examples");
+        streamManager.createStream("examples", "helloStream", streamConfig);
+}
+```
+Executing the above lines should ensure we have created a Pravega scope called `examples` and a Pravega Stream called `helloStream`.
+
+# 4.2 Create a Pravega Event Writer and write events into the stream.
+
+Let's create a Pravega Event Writer using the [EventStreamClientFactory](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/EventStreamClientFactory.html)
+
+```java
+try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope("examples",
+                ClientConfig.builder().controllerURI(controllerURI).build());
+     EventStreamWriter<String> writer = clientFactory.createEventWriter("helloStream",
+                new UTF8StringSerializer(), EventWriterConfig.builder().build())) {
+        writer.writeEvent("helloRoutingKey", "hello world!"); // write an event.
+}
+```
+The above snippet creates an Event Writer and writes an event into the Pravega stream. Note that `writeEvent()` returns a `CompletableFuture`, which can be captured for use or will be resolved when calling `flush()` or `close()`, and, if destined for the same segment, the futures write in the order `writeEvent()` is called.
+
+When instantiating the EventStreamWriter above, we passed in a [UTF8StringSerializer](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/impl/UTF8StringSerializer.java) instance. Pravega uses a [Serializer](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/stream/Serializer.html) interface in its writers and readers to simplify the act of writing and reading an object’s bytes to and from streams. The [JavaSerializer](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/impl/JavaSerializer.java) can handle any `Serializable` object.
+
+# 4.3 Create a Pravega Event Reader and read the event back from the stream.
```

Comment:
Same here as in the previous subsections.

Reply:
fixed.